components/classes/StudentListTableHead.tsx

```diff
@@ -0,0 +1,103 @@
+import { StylableFC } from "@/utils/types/common";
+import { list } from "radash";
+import { MAXIMUM_EMPTY_COLUMNS, OptionsType } from "./StudentListPrintout";
+
+/**
+ * The head area of the table shown in Student List Printout. Displays the
+ * column headers based on the given options.
+ *
+ * @param options The options for the printout.
+ *
+ * @note Not to be confused with the header area of the printout itself.
+ */
+const StudentListTableHead: StylableFC<{
+  options: OptionsType;
+}> = ({ options, style, className }) => (
+  <thead style={style} className={className}>
+    <tr>
+      {/* Class no. */}
+      {options.columns.includes("classNo") && (
+        <th className="w-12">
+          {options.language === "en-US" ? "№" : "เลขที่"}
+        </th>
+      )}
+
+      {/* Student ID */}
+      {options.columns.includes("studentID") && (
+        <th className="w-24">
+          {options.language === "en-US" ? "Student ID" : "เลขประจำตัว"}
+        </th>
+      )}
+
+      {/* Full name */}
+      {(options.columns.includes("prefix") ||
+        options.columns.includes("fullName")) && (
+        <th
+          colSpan={(() => {
+            let colSpan = 0;
+            // Prefix takes up one column.
+            if (options.columns.includes("prefix")) colSpan += 1;
+            // Full name takes up two columns.
+            if (options.columns.includes("fullName")) colSpan += 2;
+            return colSpan;
```

Comment:
The condition is easier to understand because the use of if statements instead of short circuit operators.